homeassistant/components/systemmonitor/sensor.py

```diff
@@ -108,6 +124,7 @@ class SystemMonitorSensor(Entity):
     def __init__(self, sensor_type, argument=""):
         """Initialize the sensor."""
         self._name = "{} {}".format(SENSOR_TYPES[sensor_type][0], argument)
+        self.entity_id = ENTITY_ID_FORMAT.format(sensor_type)
```

Comment:
This change will break existing sensors.

Reply:
I don't see why. The current implicit `entity_id` values for this integration are: `"sensor".{sensor_type}`. The same is still true after the change.

However, the change is required, since the `sensor_type` is apparently using the sensor name/label to generate the `entity_id` (rather then the more logical key aka `sensor_type` as I do now). That means without that change, the existing sensor `sensor.processor_use` would now suddenly be `sensor.process_use_percent` although I (on purpose) did not change the `sensor_type` value for it (to not break any setups, despite the fact that having it named `processor_use_percent`would be more consistent).

I had verified that my change did not break any of the existing sensors of that integration that I used, esp. not the one I actually touched (processor use).

@fabaff If I missed something, please let me know.

Reply:
> I had verified that my change did not break any of the existing sensors of that integration that I used, esp. not the one I actually touched (processor use).

See:

```yaml
  - platform: systemmonitor
    resources:
      - type: disk_use_percent
        arg: /home
```

will result in

```bash
Current name: sensor.disk_use_percent_home
New name:     sensor.disk_use_percent
```

Reply:
Thanks for pointing that out. Fix pushed to take the `argument` into account as well.